adapter/utils/src/test/java/org/mobilitydata/gtfsvalidator/geoutils/GeospatialUtilsImplTest.java

```diff
@@ -39,4 +53,613 @@ void distanceReferenceCheck() {
+
+        final StopTime stopTime3 = mock(StopTime.class);
+        when(stopTime3.getStopSequence()).thenReturn(3);
+        when(stopTime3.getStopId()).thenReturn(stopId3);
+
+        // shapes.txt
+        final String shapeId = "shape1";
+        final ShapePoint pt1 = mock(ShapePoint.class);
+        final ShapePoint pt2 = mock(ShapePoint.class);
+        final ShapePoint pt3 = mock(ShapePoint.class);
+        final ShapePoint pt4 = mock(ShapePoint.class);
+        final ShapePoint pt5 = mock(ShapePoint.class);
+
+        when(pt1.getShapeId()).thenReturn(shapeId);
+        when(pt2.getShapeId()).thenReturn(shapeId);
+        when(pt3.getShapeId()).thenReturn(shapeId);
+        when(pt4.getShapeId()).thenReturn(shapeId);
+        when(pt5.getShapeId()).thenReturn(shapeId);
+
+        when(pt1.getShapePtSequence()).thenReturn(1);
+        when(pt2.getShapePtSequence()).thenReturn(2);
+        when(pt3.getShapePtSequence()).thenReturn(3);
+        when(pt4.getShapePtSequence()).thenReturn(4);
+        when(pt5.getShapePtSequence()).thenReturn(5);
+
+        when(pt1.getShapePtLat()).thenReturn(28.05724310653972f);
+        when(pt1.getShapePtLon()).thenReturn(-82.41350776611507f);
+        when(pt2.getShapePtLat()).thenReturn(28.05746701492806f);
+        when(pt2.getShapePtLon()).thenReturn(-82.41493135129478f);
+        when(pt3.getShapePtLat()).thenReturn(28.05800068503469f);
+        when(pt3.getShapePtLon()).thenReturn(-82.4159394137605f);
+        when(pt4.getShapePtLat()).thenReturn(28.05808869825447f);
+        when(pt4.getShapePtLon()).thenReturn(-82.41648754043338f);
+        when(pt5.getShapePtLat()).thenReturn(28.05809979887893f);
+        when(pt5.getShapePtLon()).thenReturn(-82.41773971025437f);
+
+        // trips.txt
+        final String tripId = "trip1";
+        final Trip trip = mock(Trip.class);
+        when(trip.getTripId()).thenReturn(tripId);
+        when(trip.getShapeId()).thenReturn(shapeId);
+
+        // Map containing StopTime entities. Entities are mapped on keys from GTFS file stop_times.txt:
+        // - trip_id
+        // - stop_sequence
+        final Map<String, TreeMap<Integer, StopTime>> stopTimeCollection = new HashMap<>(1);
+        final TreeMap<Integer, StopTime> stopTimes = new TreeMap<>(Map.of(1, stopTime1, 2, stopTime2, 3, stopTime3));
+        stopTimeCollection.put(tripId, stopTimes);
+
+        // Map containing Stop entities. Entities are keyed on GTFS stops.txt stop_id
+        final Map<String, LocationBase> stopPerId = new HashMap<>(Map.of(stopId1, stop1, stopId2, stop2, stopId3, stop3));
+
+        // Entities are keyed on shape_pt_sequence of GTFS file shapes.txt
+        SortedMap<Integer, ShapePoint> points = new TreeMap<>(Map.of(1, pt1, 2, pt2, 3, pt3, 4, pt4, 5, pt5));
+
+        List<StopTooFarFromTripShape> errorList =
+                GEO_UTILS.checkStopsWithinTripShape(trip, stopTimes, points, stopPerId, new HashSet<>());
+
+        assertEquals(0, errorList.size());
+    }
 }
```

Comment:
I think that we should have tests for:
- null stop should not generate notice
- null trip should not generate notice
- null stoptimes should not generate notice
- empty stoptimes should not generate notice
- null shapePoints should not generate notice
- empty shapePoints should not generate notice